source/Xunit.Gherkin.Quick.UnitTests/GherkinScenarioOutlineTests.cs

```diff
@@ -307,12 +307,578 @@ public void ApplyExampleRow_Digests_Row_Values_Into_Scenario_With_DocString_In_S
+                            {
+                                new Gherkin.Ast.TableCell(null, "1")
+                            })
+                        })
+                });
+
+            //act.
+            var scenario = sut.ApplyExampleRow("existing example", 0);
+
+            //assert.
+            Assert.NotNull(scenario);
+            Assert.Equal(sut.Name, scenario.Name);
+
+            var scenarioSteps = scenario.Steps.ToList();
+            var sutSteps = sut.Steps.ToList();
+            
+            Assert.Equal(sutSteps.Count, scenarioSteps.Count);
+            Assert.Equal(3, scenarioSteps.Count);
+
+            // Check regular text
+            Assert.NotEqual(sutSteps[0].Text, scenarioSteps[0].Text);
+            Assert.Equal(scenarioSteps[0].Text, expectedResultSentence);
+
+            // Check DataTable
+            Assert.IsType<DataTable>(scenarioSteps[1].Argument);
+            var dataTable = (DataTable)scenarioSteps[1].Argument;
+            var rows = dataTable.Rows.ToList();
+            Assert.Equal(2, rows.Count());
+
+            var row0 = rows[0].Cells.ToArray();
+            Assert.Equal(3, row0.Count());
+            var row1 = rows[1].Cells.ToArray();
+            Assert.Equal(3, row1.Count());
+
+            Assert.Equal(row0[0].Value, column0);
+            Assert.Equal(row0[1].Value, column1);
+            Assert.Equal(row0[2].Value, column2);
+            Assert.Equal(row1[0].Value, expectedResultCell0);
+            Assert.Equal(row1[1].Value, expectedResultCell1);
+            Assert.Equal(row1[2].Value, expectedResultCell2);
+
+            // Check DocString
+            Assert.IsType<DocString>(scenarioSteps[2].Argument);
+            var docString = (DocString)scenarioSteps[2].Argument;
+
+            Assert.NotEqual(docStringText, docString.Content);
+            Assert.Equal(expectedDocStringContent, docString.Content);
+        }
+
+        [Theory]
+        [InlineData("a<a", "value is <a<a>", true, false, false)]
+        [InlineData("a<a", "value is <a<a>", false, true, false)]
+        [InlineData("a<a", "value is <a<a>", false, false, true)]
+        [InlineData("a<a", "value is <a<a>", true, true, false)]
+        [InlineData("a<a", "value is <a<a>", false, true, true)]
+        [InlineData("a<a", "value is <a<a>", true, false, true)]
+        [InlineData("a<a", "value is <a<a>", true, true, true)]
+        [InlineData("a>a", "value is <a>a>", true, false, false)]
+        [InlineData("a>a", "value is <a>a>", false, true, false)]
+        [InlineData("a>a", "value is <a>a>", false, false, true)]
+        [InlineData("a>a", "value is <a>a>", true, true, false)]
+        [InlineData("a>a", "value is <a>a>", false, true, true)]
+        [InlineData("a>a", "value is <a>a>", true, false, true)]
+        [InlineData("a>a", "value is <a>a>", true, true, true)]
+        public void DontApplyExampleRow_And_Throw_Cant_Substite_Wrong_Example_Parameter(
```

Comment:
Cases when we want to use "<" and ">" in the middle of an example's name. The parser tries to grab the correct pattern unexpectedly for the user, and the error is enough for the user to understand what the issue is.